### PR TITLE
Clean up contentAuthoring.md

### LINKED ISSUE #437: Split up Content Authoring page on documentation site
Follow up on discussion with @damithc on 19 Sep

The Content Authoring page (https://markbind.github.io/markbind/userGuide/contentAuthoring.html) is too long and covers too many different topics.

We can split it up into multiple pages, one for each set of features, for example:
- Front Matter
- Markdown reference
- Headers and Footers
etc.

### PR DESCRIPTION
**Pull request by jamos-tay**
**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**

• [X] Documentation update

Fixes #437 

**What is the rationale for this request?**

The content authoring page on markbind is disorganized. It may be helpful to group topics into specific categories.

**What changes did you make? (Give an overview)**

Here is the list of headers on the current `contentAuthoring.html`, in the order that they appear on the webpage:

Content Authoring
- Mixing HTML with Markdown
- Specifying Path Reference
- Markdown Syntax
- Markdown Variables
- Font Awesome Icons
- Glyphicons
- Timestamp
- Markbind Link
- Add your own variables
- Use Searchbar
- Keywords
- Use Components
- Question body syntax
- Include Contents
- Front Matter
- Inserting content into a page's head element
- Site Navigation
- Using Footers
- Specifying footer file in Front Matter
- Defining your own inline footer
- Heading Anchors
- Layouts

---

Here is how I propose they be grouped:

Content Authoring: Topics relating to markdown syntax
- Mixing HTML with Markdown
- Supported Markdown Syntax
- Include Contents

Markbind Components: Topics relating to custom markdown components
- Use Components
- Question body syntax
- Site Navigation
- Use Searchbar
  - Keywords

Using Variables: Topics relating to variables (`{{...}}`)
- Use MarkBind variables
- Specifying Path Reference
- Font Awesome Icons
- Glyphicons
- Timestamp
- Markbind Link
- Add your own variables

Page Layout: Topics relating to features that affect the page layout
- Front Matter
- Inserting content into a page's head element
- Using Footers
- Specifying footer file in Front Matter
- Defining your own inline footer
- Layouts

Heading Anchors: Removed - No documentation needed for this section as they are automatically generated. Maybe this feature can be shown on the landing page?

Here is how it appears (sidebar):
image

Some minor changes:

Standardized headings to level 3, `###`
Wrote a short one line introduction for pages that needed it

### CODE REVIEW

**Comment by damithc** [comment]
* The page on Netlify preview is different from other pages. Maybe redesign to match other pages?
* We only have a user guide. Should we add other pages, such as About us etc.?

**Comment by jamos-tay** [comment]
@damithc Sure - both of those are good ideas, I'll create issues for them. I think those changes should be made separately though since the goal for this PR was just to restructure the content authoring section, I didn't introduce any new content.

File: docs/_markbind/navigation/userGuideSections.md [hunk 1]
```diff
@@ -5,6 +5,9 @@
   * [Site Configuration]({{baseUrl}}/userGuide/siteConfiguration.html)
 * [Content Authoring]({{baseUrl}}/userGuide/contentAuthoring.html)
   * [Including Contents]({{baseUrl}}/userGuide/includingContents.html)
+  * [Markbind Components]({{baseUrl}}/userGuide/markbindComponents.html)
```
Comment (yamgent) [inline docs/_markbind/navigation/userGuideSections.md]: `Markbind Components` -> `Using Components`

Remember to make the changes in `header.md` and `site.json` too.

File: docs/userGuide/contentAuthoring.md [hunk 2]
```diff
@@ -119,547 +110,10 @@ In addition, it supports:
...
-```
-
-Note:
-- Footer file specified in [front matter](#front-matter) overrides all inline footers in the same page.
-- Inline footers need to be the last element in a page.
-- If there is more than one inline footer in a page, only the last inline footer will be rendered.
-- Footers should not be nested in other components or HTML tags. If found inside, they will be shifted outside to be rendered properly.
-- [MarkBind components](#use-components) and [includes](#include-contents) are not supported in footers.
-
-### Heading Anchors
```
Comment (yamgent) [inline docs/userGuide/contentAuthoring.md]: This sub-section seems to have disappeared in the cleaned up version. :P

File: docs/userGuide/usingVariables.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
+<frontmatter>
+  footer: userGuideFooter.md
+  siteNav: userGuideSections.md
+</frontmatter>
+
+<include src="../common/header.md" />
+
+<div class="website-content">
+
+# Using MarkBind Variables
```
Comment (yamgent) [inline docs/userGuide/usingVariables.md]: `# Using Variables` will do.

File: docs/userGuide/contentAuthoring.md [hunk 4]
```diff
@@ -7,11 +7,7 @@
 
 <div class="website-content">
 
-# Content authoring
-
-### General writing guide
-
-#### Mixing HTML with Markdown
+# Content Authoring
```
Comment (yamgent) [inline docs/userGuide/contentAuthoring.md]: Just a side note: Some pages seems to have caps for all words but some pages don't (see `Including Contents` and `Known Problems`)

I think in a future PR could do some more minor cleanups in the documentation.

File: docs/userGuide/contentAuthoring.md (hunk 2 above)
Comment (yamgent) [reply docs/userGuide/contentAuthoring.md]: This feature doesn't seem to be worthy enough to feature on the landing page.

I am OK with removing this section if there's no suitable place for it.

File: docs/userGuide/usingComponents.md [hunk 5]
```diff
@@ -0,0 +1,208 @@
+<frontmatter>
+  footer: userGuideFooter.md
+  siteNav: userGuideSections.md
+</frontmatter>
+
+<include src="../common/header.md" />
+
+<div class="website-content">
+
+# Components
```
Comment (yamgent) [inline docs/userGuide/usingComponents.md]: For consistency, the title should be `# Using Components`

**Comment by jamos-tay** [comment]
Updated

**Pull request merged**